ethereum/statetransition/src/main/java/tech/pegasys/artemis/statetransition/util/EpochProcessorUtil.java

```diff
@@ -655,13 +614,13 @@ public static void update_validator_registry(BeaconState state) throws EpochProc
    */
   public static void currentStateUpdatesAlt1(BeaconState state) throws EpochProcessingException {
     try {
-      UnsignedLong epoch = BeaconStateUtil.get_next_epoch(state);
+      long epoch = BeaconStateUtil.get_next_epoch(state);
       state.setCurrent_shuffling_epoch(epoch);
 
-      UnsignedLong SHARD_COUNT = UnsignedLong.valueOf(Constants.SHARD_COUNT);
-      UnsignedLong committee_count = BeaconStateUtil.get_current_epoch_committee_count(state);
-      UnsignedLong current_epoch_start_shard =
-          state.getCurrent_shuffling_start_shard().plus(committee_count).mod(SHARD_COUNT);
+      long SHARD_COUNT = Constants.SHARD_COUNT;
+      long committee_count = BeaconStateUtil.get_current_epoch_committee_count(state);
+      long current_epoch_start_shard =
+          state.getCurrent_shuffling_start_shard() + committee_count % SHARD_COUNT;
```

Comment:
The order of operations in the UnsignedLong version was:
 `(state.getCurrent_shuffling_start_shard() + committee_count) % SHARD_COUNT;`

Can we double check this?